src/python/pants/backend/java/test/junit.py

```diff
@@ -57,13 +57,24 @@ async def run_junit_test(
             prefix="__thirdpartycp",
             lockfiles=(lockfile,),
             maven_requirements=(
-                MavenRequirements.create_from_maven_coordinates_fields(
-                    fields=(),
-                    additional_requirements=[
-                        "org.junit.platform:junit-platform-console:1.7.2",
-                        "org.junit.jupiter:junit-jupiter-engine:5.7.2",
-                        "org.junit.vintage:junit-vintage-engine:5.7.2",
-                    ],
+                JvmCoordinates(
+                    [
+                        Coordinate(
```

Comment:
The `:`-separated strings are still in pretty common usage in gradle land, and provide dictionary-style notation as a newer option:

```
// https://mvnrepository.com/artifact/org.junit.platform/junit-platform-console
testImplementation group: 'org.junit.platform', name: 'junit-platform-console', version: '1.8.0'
```

or 
```
// https://mvnrepository.com/artifact/org.junit.platform/junit-platform-console
testImplementation 'org.junit.platform:junit-platform-console:1.8.0'
```

Supporting the string-style coordinates would probably be a good idea.

Reply:
Other prior art:

- Scala splits them but in an unstructured way: `libraryDependencies += "org.apache.derby" % "derby" % "10.4.1.3"` [Scala docs](https://www.scala-sbt.org/1.x/docs/Library-Dependencies.html#Managed+Dependencies)
- Bazel's [`rules_jvm_external`](https://github.com/bazelbuild/rules_jvm_external) uses colon-delimited strings for the `maven_install` rule.
- Clojure uses [a LISPish custom DSL](https://clojure.org/guides/deps_and_cli): `{clojure.java-time/clojure.java-time {:mvn/version "0.3.2"}}}`